ujson/src/ujson/Readable.scala

```diff
@@ -15,14 +15,15 @@ object Readable extends ReadableLowPri{
   }
   implicit def fromString(s: String) = new fromTransformer(s, StringParser)
   implicit def fromCharSequence(s: CharSequence) = new fromTransformer(s, CharSequenceParser)
-  implicit def fromPath(s: java.nio.file.Path) = new fromTransformer(
-    java.nio.file.Files.newInputStream(s),
-    InputStreamParser
-  )
-  implicit def fromFile(s: java.io.File) = new fromTransformer(
-    java.nio.file.Files.newInputStream(s.toPath),
-    InputStreamParser
-  )
+  implicit def fromPath(s: java.nio.file.Path) = {
+    val inputStream = java.nio.file.Files.newInputStream(s)
+    new fromTransformer(inputStream, InputStreamParser) {
+      override def transform[T](f: Visitor[_, T]) =
```

Comment:
@nafg since we are overriding `def transform`, could we skip `fromTransformer` entirely and instead inherit directly from `Readable`? That would allow us to put the `val inputStream` inside the `def transform` function, making it open a new `java.io.InputStream` every time `transform` is called

Reply:
I don't see why not, could you take care of it?